docs/contributor/04-10-technical-design.md

```diff
@@ -146,21 +146,20 @@ For now, the following scenarios must be covered by this component:
 - Restart Pods with proxy sidecar when CNI config changes.
 - Restart Pods with proxy sidecar after an Istio version update.
 - Restart Pods with proxy sidecar when proxy resources change.
+- Restart Pods if they match predicates that IstioResourcesReconciliation component will specify (e.g. being up-to-date with an EnvoyFilter)
 
-#### IstioIngressGatewayReconciliation
+#### IstioResourcesReconciliation
 
-The IstioIngressGatewayReconciliation component is responsible for bringing Istio Ingress Gateway to the desired state.
+IstioResourcesReconciliation is a component responsible for applying resources dependent on Istio (e.g. VirtualService, EnvoyFilter) and making sure that the state of Istio service-mesh is configured according to those resources.
```

Comment:
We are missing the info about the annotation we are using in the EnvoyFilter.
I think we might want to have it more general mentioned in the technical design, that for configurations that we cannot directly check on the resources itself, we use this annotation.
In the ADR I think it might be even better to use the concrete example of the EnvoyFilter and explaining why we chose the annotation over other options, e.g. reading the configuration directly.
So the ADR is our record to understand in the future why we used the annotation-based approach instead of istiod.